pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -140,69 +140,68 @@ func (r *RootSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, status.APIServerError(err, "failed to get RootSync")
 	}
 
+	currentRS := rs.DeepCopy()
 	if !rs.DeletionTimestamp.IsZero() {
 		// Deletion requested.
 		// Cleanup is handled above, after the RootSync is NotFound.
 		r.logger(ctx).V(3).Info("Sync deletion timestamp detected")
-	}
-
-	currentRS := rs.DeepCopy()
-
-	if err := r.validateNamespaceName(rsRef.Namespace); err != nil {
-		r.logger(ctx).Error(err, "Sync namespace invalid")
-		rootsync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
-		return controllerruntime.Result{}, updateErr
-	}
-
-	if errs := validation.IsDNS1123Subdomain(reconcilerRef.Name); errs != nil {
-		err := errors.Errorf("Invalid reconciler name %q: %s.", reconcilerRef.Name, strings.Join(errs, ", "))
-		r.logger(ctx).Error(err, "Sync name or namespace invalid")
-		rootsync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
-		return controllerruntime.Result{}, updateErr
-	}
-
-	if err := r.validateSpec(ctx, rs, reconcilerRef.Name); err != nil {
-		r.logger(ctx).Error(err, "Sync spec invalid")
-		rootsync.SetStalled(rs, "Validation", err)
-		// Validation errors should not trigger retry (return error),
-		// unless the status update also fails.
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		// Use the validation error for metric tagging.
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
-		return controllerruntime.Result{}, updateErr
-	}
+	} else {
```

Comment:
This is a pretty long nested conditional, could this be refactored for readability?

Reply:
refactored